base/game_events/GameEventManager.ts

```diff
@@ -441,6 +442,8 @@ export class GameEventManager {
                     info.pos,
                     info.collision_layers
                 );
+            case event_types.DESTROYER:
+                return new DestroyerEvent(this.game, this.data, info.active, info.key_name);
```

Comment:
Once you changed the constructor in the Destroyer class, also add `target_event_key` in the parameters list.

Reply:
and should the target_event_key come from the info object?

Reply:
Yes, exactly! `info.target_event_key `